backend/post-service/src/main/java/com/speakapp/postservice/repositories/CommentRepository.java

```diff
@@ -14,7 +16,18 @@
 public interface CommentRepository extends JpaRepository<Comment, UUID> {
 
     Page<Comment> findAllByPostOrderByCreatedAtDesc(Post post, Pageable pageable);
+    Page<Comment> findAllByPostOrderByCreatedAtAsc(Post post, Pageable pageable);
 
     List<Comment> findAllByPostOrderByCreatedAtDesc(Post post);
 
+    @Query("SELECT c FROM Comment c LEFT JOIN CommentReaction cr ON c.commentId = cr.comment.commentId " +
+            "WHERE c.post = :post " +
+            "GROUP BY c.commentId " +
+            "ORDER BY COUNT(cr) DESC, c.createdAt DESC")
+    Page<Comment> findAllByPostOrderByReactionsDesc(@Param("post") Post post, Pageable pageable);
```

Comment:
nie wystarczy po prostu dodać parametru do `PageRequest.of`?
Takie coś jest na baeldung:
``` 
Pageable sortedByName = 
  PageRequest.of(0, 3, Sort.by("name"));

Pageable sortedByPriceDesc = 
  PageRequest.of(0, 3, Sort.by("price").descending());

Pageable sortedByPriceDescNameAsc = 
  PageRequest.of(0, 5, Sort.by("price").descending().and(Sort.by("name")));
```
https://www.baeldung.com/spring-data-jpa-pagination-sorting

Reply:
Jeśli dodanie w komentarzu pola z ilością reakcji by ułatwiło to wszystko, to może dobrze by było to zrobić? Tylko wtedy trzeba zadbać o odpowiednie aktualizowanie go.